### ISSUE TITLE
site: Potential UnicodeDecodeError when handling pth file

### ISSUE THREAD

**Issue by methane** [python/cpython#87380]
BPO | [43214](https://bugs.python.org/issue43214)
--- | :---
Nosy | @vstinner, @methane
PRs | <li>python/cpython#24837</li>
Superseder | <li>bpo-37364: Use io.open_code for .pth files</li>

<sup>*Note: these values reflect the state of the issue at the time it was migrated and might not reflect the current state.*</sup>

<details><summary>Show more details</summary><p>

GitHub fields:
```python
assignee = None
closed_at = <Date 2021-03-17.08:54:25.835>
created_at = <Date 2021-02-13.08:53:26.246>
labels = ['easy', 'type-bug', '3.8', '3.9', '3.10', 'invalid', 'library']
title = 'site: Potential UnicodeDecodeError when handling pth file'
updated_at = <Date 2021-03-17.10:30:46.450>
user = 'https://github.com/methane'
```

bugs.python.org fields:
```python
activity = <Date 2021-03-17.10:30:46.450>
actor = 'vstinner'
assignee = 'none'
closed = True
closed_date = <Date 2021-03-17.08:54:25.835>
closer = 'methane'
components = ['Library (Lib)']
creation = <Date 2021-02-13.08:53:26.246>
creator = 'methane'
dependencies = []
files = []
hgrepos = []
issue_num = 43214
keywords = ['patch', 'easy', '3.8regression']
message_count = 3.0
messages = ['386916', '388911', '388916']
nosy_count = 2.0
nosy_names = ['vstinner', 'methane']
pr_nums = ['24837']
priority = 'normal'
resolution = 'not a bug'
stage = 'resolved'
status = 'closed'
superseder = '37364'
type = 'behavior'
url = 'https://bugs.python.org/issue43214'
versions = ['Python 3.8', 'Python 3.9', 'Python 3.10']
```

</p></details>

**Comment by wkwkw** [comment]
I'm having problem with UnicodeDecodeError in this situation.
I know it can be avoided using utf-8 mode, but utf-8 mode is not working with 3.11 nor 3.12 in my environment: 
* Windows 11
* python 3.11.8 / 3.12.2

I'd appreciate if pth file encoding be utf8, or someone knows a workaround to make utf-8 mode available in python 3.11 and later.
(with python 3.10 or earlier, utf-8 mode is working fine.)

I'm using new project and package manager [rye](https://rye-up.com/) that creates a pth file that contains full path of the project, which often contains non-ascii characters in some countries. So this could happen more often when rye would get more popular I guess.

**Comment by vstinner** [comment]
I suggest you to open a new issue since this one is closed.
